src/transformers/models/mgp_str/processing_mgp_str.py

```diff
@@ -0,0 +1,185 @@
+# coding=utf-8
+# Copyright 2022 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Processor class for MGP-STR."""
+
+from transformers import BertTokenizer, GPT2Tokenizer, MGPSTRTokenizer
+
+from ...processing_utils import ProcessorMixin
+
+
+class MGPSTRProcessor(ProcessorMixin):
+    r"""
+    Constructs a MGP-STR processor which wraps a vision feature extractor and MGP-STR tokenizers into a single
+    processor.
+
+    [`MGPSTRProcessor`] offers all the functionalities of `ViTFeatureExtractor`] and [`MGPSTRTokenizer`]. See the
+    [`~MGPSTRProcessor.__call__`] and [`~MGPSTRProcessor.batch_decode`] for more information.
+
+    Args:
+        feature_extractor (`ViTFeatureExtractor`):
+            An instance of `ViTFeatureExtractor`. The feature extractor is a required input.
+    """
+    feature_extractor_class = "AutoFeatureExtractor"
+    tokenizer_class = "AutoTokenizer"
+
+    def __init__(self, feature_extractor, *args):
```

Comment:
The other processor classes allow the user to pass in any of the specific modality processing classes to create the class e.g. `def __init__(self, image_processor, tokenizer)` [in XClip](https://github.com/huggingface/transformers/blob/0db5d911fc94604f9568b4b212e005ec4600d157/src/transformers/models/x_clip/processing_x_clip.py#L42). This allows for fully configurability from the user and ensures the checkpoints used for processing e.g. `"bery-base-uncased"` are recorded in the preprocessing config. Could you add to this class so this is possible?